closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -362,6 +377,46 @@ int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         return (len + tagLenBytes);
     }
 
+    /**
+     * Performs encryption operation for the last time.
+     *
+     * @param src the input buffer with the data to be encrypted
+     * @param dst the offset in <code>in</code>
+     * @return the number of bytes placed into the <code>out</code> buffer
+     */
+    int encryptFinal(ByteBuffer src, ByteBuffer dst)
+        throws IllegalBlockSizeException, ShortBufferException {
+
+        // Get array from source
+        byte[] src_arr = getBbArray(src);
+        int src_ofs = src.hasArray() ? (src.position() + src.arrayOffset()) : 0;
+
+        // Get array from destination
+        byte[] dst_arr;
+        int dst_offset;
+        if (dst.hasArray()) {
+            dst_arr = dst.array();
+            dst_offset = dst.position() + dst.arrayOffset();
+        } else {
+            dst_arr = new byte[dst.capacity()];
```

Comment:
This is larger than necessary; it should be:
```
dst_arr = new byte[dst.remaining()];
```

Reply:
Fixed.